### PR TITLE
844883 - [Others] UG Corrections.

### PR DESCRIPTION
**Pull request by sudarsanMS**
**Description**
- Included the auto generate columns for custom type topic in data binding.
- Modified the code alignment, images and item source properly for all the topics.
- Included the conditional styling of all summary cells using converter in conditional styling.
- Included the Binding complex properties topic in data binding.
- Updated the upcoming features.

### CODE REVIEW

File: MAUI/DataGrid/conditional-styling.md [hunk 1]
```diff
@@ -382,4 +385,196 @@ public class ForeColorConverter : IValueConverter
...
+
+        //custom condition is checked.
+
+        if ((double)calculatedValue < 45000)
+            return Colors.LightGreen;
+        return Colors.LightBlue;
+    }
+    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        throw new NotImplementedException();
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/conditional-styling.md:438]: Return null instead through exception.

File: MAUI/DataGrid/conditional-styling.md [hunk 2]
```diff
@@ -382,4 +385,196 @@ public class ForeColorConverter : IValueConverter
...
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+Here, caption summary cells are customized based on `Salary` summary value.
+
+![Conditional Style of DataGrid Caption Summary Cell using Converter](maui-datagrid-conditional-captionsummary.png)
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/conditional-styling.md:446]: Conditional styling of caption summary cell using converter.

File: MAUI/DataGrid/conditional-styling.md [hunk 3]
```diff
@@ -382,4 +385,196 @@ public class ForeColorConverter : IValueConverter
...
+
+        //custom condition is checked.
+
+        if ((double)calculatedValue % 2 == 0)
+            return Colors.LightGreen;
+        return Colors.LightBlue;
+    }
+    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        throw new NotImplementedException();
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/conditional-styling.md:499]: Return null instead through exception.

File: MAUI/DataGrid/conditional-styling.md [hunk 4]
```diff
@@ -382,4 +385,196 @@ public class ForeColorConverter : IValueConverter
...
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+Here, group summary cells are customized based on `Salary` summary value.
+
+![Conditional Style of DataGrid Group Summary Cell using Converter](maui-datagrid-conditional-groupsummary.png)
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/conditional-styling.md:507]: Conditional styling of group summary cell using converter.

File: MAUI/DataGrid/conditional-styling.md [hunk 5]
```diff
@@ -382,4 +385,196 @@ public class ForeColorConverter : IValueConverter
...
+
+        //custom condition is checked.
+
+        if ((double)calculatedValue % 2 == 0)
+            return Colors.LightGreen;
+        return Colors.LightBlue;
+    }
+    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+        throw new NotImplementedException();
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/conditional-styling.md:572]: Return null instead through exception.

File: MAUI/DataGrid/conditional-styling.md [hunk 6]
```diff
@@ -382,4 +385,196 @@ public class ForeColorConverter : IValueConverter
...
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+Here, table summary cells are customized based on `Salary` summary value.
+
+![Conditional Style of WPF DataGrid Table Summary Cells using Converter](maui-datagrid-conditional-tablesummary.png)
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/conditional-styling.md:580]: Conditional styling of table summary cell using converter.

File: MAUI/DataGrid/conditional-styling.md [hunk 7]
```diff
@@ -382,4 +385,196 @@ public class ForeColorConverter : IValueConverter
...
+### Conditional styling of table summary cells using converter
+
+The appearance of table summary cell can be customized conditionally based on summary value using `converter`, where converter returns the value based on summary value. 
+
+{% tabs %}
+{% highlight xaml %}
+    <ContentPage.Resources>
+        <local:ColorConverter x:Key="converter" />
+        <Style TargetType="syncfusion:DataGridTableSummaryCell">
+        <Setter Property="Background"
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/conditional-styling.md:522]: Align setter tag properly.

File: MAUI/DataGrid/conditional-styling.md [hunk 8]
```diff
@@ -382,4 +385,196 @@ public class ForeColorConverter : IValueConverter
...
+### Conditional styling of group summary cell using converter
+
+The appearance of group summary cell can be customized conditionally based on summary value by using `converter`, where converter returns the value based on summary value.
+
+{% tabs %}
+{% highlight xaml %}
+    <ContentPage.Resources>
+        <local:ColorConverter x:Key="converter" />
+        <Style TargetType="syncfusion:DataGridGroupSummaryCell">
+        <Setter Property="Background"
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/conditional-styling.md:461]: Align setter tag properly.

File: MAUI/DataGrid/conditional-styling.md [hunk 9]
```diff
@@ -382,4 +385,196 @@ public class ForeColorConverter : IValueConverter
...
+
+### Conditional styling of caption summary cells using converter
+
+The appearance of caption summary cell can be customized conditionally based on [SummaryValues]() by using `converter`, where converter returns the value based on `SummaryValues`. 
+
+{% tabs %}
+{% highlight xaml %}
+    <ContentPage.Resources>
+        <local:ColorConverter x:Key="converter" />
+        <Style TargetType="syncfusion:DataGridCaptionSummaryCell">
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/conditional-styling.md:400]: Align setter tag properly.

File: MAUI/DataGrid/selection.md [hunk 10]
```diff
@@ -378,6 +378,27 @@ private void dataGrid_CellTapped(object sender,DataGridCellTappedEventArgs e)
 {% endhighlight %}
 {% endtabs %}
 
+### CellRightTapped event
```
[the remark this answers is not in the document]
Comment (sudarsanMS) [reply MAUI/DataGrid/selection.md:381]: removed as we discussed..

File: MAUI/DataGrid/overview.md [hunk 11]
```diff
@@ -41,4 +41,6 @@ The .NET MAUI DataGrid control is used to display and manipulate data in a tabul
 
 * **Stacked Headers** - Extensive support to show multiple headers called stacked headers.
 
-* **Exporting** - Support to export the data to Excel and PDF.
\ No newline at end of file
+* **Exporting** - Support to export the data to Excel and PDF.
+
+N> You can refer to our [MAUI DataGrid](https://www.syncfusion.com/maui-controls/maui-datagrid) feature tour page for its groundbreaking feature representations. You can also explore our [MAUI DataGrid example](https://github.com/syncfusion/maui-demos/tree/master/MAUI/DataGrid) to knows various chart types and how to easily configured with built-in support for creating stunning visual effects.
```
[the remark this answers is not in the document]
Comment (sudarsanMS) [reply MAUI/DataGrid/overview.md:46]: Modified

File: MAUI/DataGrid/migration.md [hunk 12]
```diff
@@ -41,6 +41,11 @@ To make migration from [Xamarin SfDataGrid](https://www.syncfusion.com/xamarin-u
 <td>{{'[DataGridSortingGestureType](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridSortingGestureType.html)'| markdownify }}</td>
 <td>Defines the constants that specifies how the sorting is applied.</td>
 </tr>
+<tr>
+<td>{{'[Position](https://help.syncfusion.com/cr/xamarin/Syncfusion.SfDataGrid.XForms.GridTableSummaryRow.html#Syncfusion_SfDataGrid_XForms_GridTableSummaryRow_PositionProperty)'| markdownify }}</td>
+<td>{{'[SummaryRowPosition]()'| markdownify }}</td>
```
[the remark this answers is not in the document]
Comment (sudarsanMS) [reply MAUI/DataGrid/migration.md:46]: Added

File: MAUI/DataGrid/grouping.md [hunk 13]
```diff
@@ -350,12 +350,12 @@ this.dataGrid.AllowGroupExpandCollapse = true;
 
 ### Expand or collapse all the groups
 
-To expand and collapse the groups programmatically, you can simply invoke the [SfDataGrid.ExpandAllGroup](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_ExpandAllGroups) and [SfDataGrid.CollapseAllGroup]() methods.
+To expand and collapse the groups programmatically, you can simply invoke the [SfDataGrid.ExpandAllGroups](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.SfDataGrid.html#Syncfusion_Maui_DataGrid_SfDataGrid_ExpandAllGroups) and [SfDataGrid.CollapseAllGroups]() methods.
```
[the remark this answers is not in the document]
Comment (sudarsanMS) [reply MAUI/DataGrid/grouping.md:353]: Added

File: MAUI/DataGrid/migration.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (sudarsanMS) [reply MAUI/DataGrid/migration.md:46]: Added

File: MAUI/DataGrid/data-binding.md [hunk 14]
```diff
@@ -61,6 +62,27 @@ this.sfDataGrid1.ItemsSource = table;
 * [SfDataGrid.View.Filter](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Data.ICollectionViewAdv.html#Syncfusion_Maui_Data_ICollectionViewAdv_Filter) is not supported.
 * [SfDataGrid.View.LiveDataUpdateMode](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Data.ICollectionViewAdv.html#Syncfusion_Maui_Data_ICollectionViewAdv_LiveDataUpdateMode) is not supported.
 
+## Binding Complex properties
```
[the remark this answers is not in the document]
Comment (sudarsanMS) [reply MAUI/DataGrid/data-binding.md:65]: changed

File: MAUI/DataGrid/conditional-styling.md (hunk 9 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/conditional-styling.md:400]: modified

File: MAUI/DataGrid/conditional-styling.md (hunk 8 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/conditional-styling.md:461]: modified

File: MAUI/DataGrid/conditional-styling.md (hunk 7 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/conditional-styling.md:522]: modified

File: MAUI/DataGrid/conditional-styling.md (hunk 6 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/conditional-styling.md]: Changed

File: MAUI/DataGrid/columns.md [hunk 15]
```diff
@@ -84,6 +84,26 @@ Similarly, it retains the sorting of the columns that are defined explicitly at
 </tr>
 </table>
 
+### Auto generate columns for custom type
+
+By default columns are also auto generated for custom type properties and for parent properties of complex properties in the data object.
```
[the remark this answers is not in the document]
Comment (sudarsanMS) [reply MAUI/DataGrid/columns.md:89]: Modified

File: MAUI/DataGrid/conditional-styling.md (hunk 1 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/conditional-styling.md:438]: modified

File: MAUI/DataGrid/conditional-styling.md (hunk 2 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/conditional-styling.md:446]: modified

File: MAUI/DataGrid/conditional-styling.md (hunk 3 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/conditional-styling.md:499]: modified

File: MAUI/DataGrid/conditional-styling.md (hunk 4 above)
Comment (sudarsanMS) [reply MAUI/DataGrid/conditional-styling.md:507]: modified

File: MAUI/DataGrid/data-binding.md [hunk 16]
```diff
@@ -61,6 +62,27 @@ this.sfDataGrid1.ItemsSource = table;
 * [SfDataGrid.View.Filter](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Data.ICollectionViewAdv.html#Syncfusion_Maui_Data_ICollectionViewAdv_Filter) is not supported.
 * [SfDataGrid.View.LiveDataUpdateMode](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Data.ICollectionViewAdv.html#Syncfusion_Maui_Data_ICollectionViewAdv_LiveDataUpdateMode) is not supported.
 
+## Binding Complex properties
+
+SfDataGrid control provides support to bind complex property to its columns. To bind the complex property to [DataGridColumn](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridColumn.html), set the complex property path to [MappingName](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataGrid.DataGridColumn.html#Syncfusion_Maui_DataGrid_DataGridColumn_MappingName).
```
[the remark this answers is not in the document]
Comment (sudarsanMS) [reply MAUI/DataGrid/data-binding.md]: modified

**Review by FarjanaParveen — APPROVED** [review]
This is fine

**Pull request merged**